viz_scripts/generic_timeseries.ipynb

```diff
@@ -154,26 +157,48 @@
+    "            # Sum daily distance traveled for each mode\n",
+    "            mode_distance = data.groupby(['user_id','date_time','Mode_confirm'], as_index=False)[['distance']].sum()\n",
+    "            mode_distance.rename(columns={'sum':'distance'}, inplace=True)\n",
+    "            mode_distance['distance'] = mode_distance['distance'].fillna(0)\n",
+    "\n",
+    "            # Sum daily emissions for each user\n",
+    "            emissions = data.groupby(['user_id','date_time'], as_index=False)[['Mode_confirm_kg_CO2', 'distance']].sum()\n",
+    "            emissions['Mode_confirm_kg_CO2'] = emissions['Mode_confirm_kg_CO2'].fillna(0)\n",
+    "            emissions['distance'] = emissions['Mode_confirm_kg_CO2'].fillna(0)\n",
+    "\n",
+    "            # Sum daily energy for each user\n",
+    "            energy = data.groupby(['user_id','date_time'], as_index=False)[['Mode_confirm_EI(kWH)', 'distance']].sum()\n",
+    "            energy['Mode_confirm_EI(kWH)'] = energy['Mode_confirm_EI(kWH)'].fillna(0)\n",
+    "            energy['distance'] = energy['Mode_confirm_EI(kWH)'].fillna(0)\n",
+    "\n",
+    "            # Add 7-day rolling avg smoothing to better see trends\n",
+    "            mode_counts['trip_count_smooth'] = mode_counts.groupby(['user_id','Mode_confirm'])['trip_count'].apply(lambda x: x.rolling(7,1).mean())\n",
+    "            mode_distance['distance_smooth'] = mode_distance.groupby(['user_id','Mode_confirm'])['distance'].apply(lambda x: x.rolling(7,1).mean())\n",
+    "            emissions['distance_smooth'] = emissions.groupby(['user_id'])['distance'].apply(lambda x: x.rolling(7,1).mean())\n",
+    "            energy['distance_smooth'] = energy.groupby(['user_id'])['distance'].apply(lambda x: x.rolling(7,1).mean())\n",
+    "        else:\n",
+    "            # Sum daily distance traveled for each mode\n",
+    "            mode_distance = data.groupby(['user_id','date_time','Mode_confirm'], as_index=False)[['distance_miles']].sum()\n",
+    "            mode_distance.rename(columns={'sum':'distance_miles'}, inplace=True)\n",
+    "            mode_distance['distance_miles'] = mode_distance['distance_miles'].fillna(0)\n",
+    "\n",
+    "            # Sum daily emissions for each user\n",
+    "            emissions = data.groupby(['user_id','date_time'], as_index=False)[['Mode_confirm_lb_CO2', 'distance_miles']].sum()\n",
+    "            emissions['Mode_confirm_lb_CO2'] = emissions['Mode_confirm_lb_CO2'].fillna(0)\n",
+    "            emissions['distance_miles'] = emissions['Mode_confirm_lb_CO2'].fillna(0)\n",
+    "\n",
+    "            # Sum daily energy for each user\n",
+    "            energy = data.groupby(['user_id','date_time'], as_index=False)[['Mode_confirm_EI(kWH)', 'distance_miles']].sum()\n",
+    "            energy['Mode_confirm_EI(kWH)'] = energy['Mode_confirm_EI(kWH)'].fillna(0)\n",
+    "            energy['distance_miles'] = energy['Mode_confirm_EI(kWH)'].fillna(0)\n",
```

Comment:
Again, why is this just copy-pasted from the previous code?

Reply:
In the above case, where dynamic config is available, we use Mode_confirm_kg_CO2 and distance (in km), while in this else case, we use Mode_confirm_lb_CO2 and distance_miles.
It's identical but different.

Reply:
The energy computation block will only be kept as a part of the default mapping case. It will be removed from the previous block.

Reply:
why? Why wouldn't you want to have the energy computation block as part of the custom mapping as well?
How else would you know whether the energy computation is correct in the custom mapping case?

Reply:
nvm, I see that you we don't have the energy computations for dynamic labels now. However, we will hopefully do so when we rewrite the calculations:
https://github.com/e-mission/e-mission-docs/issues/954
and
https://github.com/e-mission/e-mission-docs/issues/1007

I still don't see what that code has to be inline instead of a separate function that you invoke properly (potentially only when a DEBUG flag is set). That was what I had suggested earlier - if you look through the giant history here, you will find it.

Reply:
The case of `default mapping` was in-line before my code changes. I just added identical case for `dynamic config` in which distance would be in `km` instead of `mile`, and `kg` instead of `lb`. You had suggested earlier not to refactor code unnecessarily. 
The DEBUG flag implementation has been done to call the block of print/computation for CO2 which is specifically required for debugging. But this block of code is for `default mapping` case which has been handled as with an `else` part of `if len(dynamic_labels) > 0`.